c7n/resources/ecs.py

```diff
@@ -572,13 +572,15 @@ class TagEcsResource(Tag):
 
     def process_resource_set(self, client, resources, tags):
         mid = self.manager.resource_type.id
-        for r in resources:
-            if not ecs_taggable(self.manager.resource_type, r):
-                raise PolicyExecutionError("New format ecs arn required: " + r[mid])
-
         tags = [{'key': t['Key'], 'value': t['Value']} for t in tags]
+        old_arns = 0
         for r in resources:
+            if not ecs_taggable(self.manager.resource_type, r):
+                old_arns += 1
+                continue
             client.tag_resource(resourceArn=r[mid], tags=tags)
+        if old_arns != 0:
+            self.log.info("Couldn't tag %d resource(s). Needs new ARN format" % old_arns)
```

Comment:
Would recommend using new style string formatting method. This is what's the recommended standard Python string formatting for awhile, as the style you used is technically "deprecated" in favor of the following:

```
self.log.info("Couldn't tag {0} resource(s). Needs new ARN format".format(old_arns))
```

Reply:
I also don't know what the standard in the project is, but I would think that logging level should be raised to "warn" or something off of my first impression, as it's communicating a problem that's prohibiting resources from being tagged as desired from my understanding. @thisisshi could chime in on if this is against a norm in the project though.

Reply:
It also might be helpful if there was a separate log line for each resource that couldn't be tagged? that way you could actually give the resource name or something so that someone is able to tell from the logs what resources to address without further debugging/re-running. But again I'd ask @thisisshi to chime in on thoughts on this

Reply:
> Would recommend using new style string formatting method. This is what's the recommended standard Python string formatting for awhile, as the style you used is technically "deprecated" in favor of the following:

most of the project seems to use the old format, although i'm happy to change it if requested

Reply:
> It also might be helpful if there was a separate log line for each resource that couldn't be tagged?

that makes sense to me

Reply:
classic python string interpolation is not deprecated afaik (aka citation needed), nor are format() strings in any commonly used language style guide afaik, shiny for the sake of shiny isn't per se valuable, ie. needs rich formatting is a reason, else its just writing more characters to do the same thing.

Reply:
also individual logging of each resource re those that couldn't be X'd is frankly an anti pattern at this point, at scale logging a line per resource, leads to voluminous log outputs no can or wants to read, and leads to behaviors of not looking at logs. logs are not a record format without structured logging, nor are they metrics (logstash regexes not withstanding). eod, failures should be tracked separately, and there's a separate proposal issue open for that if you'd like to comment, but that shouldn't be a block to prs in progress, till its implemented.

Reply:
this should be logging at `warn` instead of info, implicit filtering in actions should always be a warning.

Reply:
changed to warn